docs/contributor/arch/016-transform-filter-support-for-telemetry-pipelines.md

```diff
@@ -0,0 +1,118 @@
+- `propagate` : The processor returns the error up the pipeline. This will result in the payload being dropped from the collector.
+
+The recommended error mode is `ignore` and this will be used as default configuration.
+
+The OTTL context will be embedded in the OTTL statements, [this is still in progress ](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/29017) and will be available in the upcoming beta version. The solution will not implement the context as configuration parameter.
+
+To ensure data consistency and sampling efficiency, the custom OTTL transformation and filtering processor will be end of the pipeline chain before exporters.
+
+The OTel configuration :
+
+```yaml
+service:
+    pipelines:
+        metrics/test-output:
+            receivers:
+                - routing/test
+                - forward/test
+            processors:
+                - filter/drop-if-input-source-runtime
+                - filter/drop-if-input-source-prometheus
+                - filter/drop-if-input-source-istio
+                - transform/set-instrumentation-scope-kyma
+                - resource/insert-cluster-name
+                - resource/delete-skip-enrichment-attribute
+                - transform/custom-transform-processor
+                - filter/custom-filter-processor
+                - batch
+            exporters:
+                - otlp/test
+```
+
+### Risk
+
+The OTTL library is still in its alpha phase, meaning that updates to the package may introduce breaking changes, including syntax and semantic updates or bugs. This poses a risk of disrupting running pipelines.
+
+### Mitigation
+
+To mitigate this risk, we suggest the following measures:
+- The [beta version](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28892) of the OTTL library will be used in the OTel Collector to ensure that the library is stable and reliable.
+- New Versions with Breaking Changes: New versions with [breaking changes](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/coding-guidelines.md#breaking-changes) of the OTTL library will be rolled out with backward compatibility. This will help users adapt changes and reduce the chance of widespread disruption.
+- Thorough Testing: We will maintain strict version control of the OTTL library, ensuring that each update is thoroughly tested. Automated tests will validate that existing pipelines continue to work as expected.
+- Documentation: We will provide detailed documentation and practical examples of how to configure filter and transformation processors using OTTL expressions.
+- Community Support: We will actively engage with the OpenTelemetry community to address any issues or concerns that arise from using the OTTL library.
+- The public Go API: The breaking changes most likely will not follow deprecation policy, this may require changes in our implementation.
```

Comment:
```suggestion
- The public Go API: The breaking changes most likely will not follow deprecation policy. This may require changes in our implementation.
```
Not sure how this is a mitigation measure; it's just a statement. Why will the breaking changes (the ones from line 114, I assume?) not follow (which?) deprecation strategy? Which specific mitigative action does this require? Something like "We must design our implementation to cope with that." or "We must continuously monitor this and change our implementation accordingly"? or something similar?

Reply:
I removed this point from the mitigation, this is more a exception to mention